hashintel/hash
# Autofocus on email input field when on the subscribe route

## 🌟 What is the purpose of this PR?




This PR improves the UX of clicking the "subscribe to mailing list" button by autofocusing the email field when landing on the subscribe route

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1202094039975696/f) _(internal)_

## 🚫 Blocked by




N/A

## 🔍 What does this change?




- Adds a check when loading the page, and when switching hash route, on whether you're on the "subscribe" hash, and autofocuses the email input field when the check passes

## 📜 Does this require a change to the docs?



No

## ⚠️ Known issues



N/A

## 🐾 Next steps



N/A

## 🛡 What tests cover this?



None

## ❓ How to test this?



1.  Click the "Join the mailing list" button and ensure the input field is focused
1.  Refresh the page while still on the `#subscribe` hash and ensure the input field remains focused
